src/libs/ajax.js

```diff
@@ -475,36 +483,75 @@ export const Jupyter = {
     return {
       localize: files => {
         return fetchLeo(`${root}/api/localize`,
-          _.mergeAll([authOpts(), jsonBody(files), { method: 'POST' }]))
+          _.mergeAll([authOpts(), jsonBody(files), { signal, method: 'POST' }]))
       },
 
       setCookie: () => {
         return fetchLeo(`${root}/setCookie`,
-          _.merge(authOpts(), { credentials: 'include' }))
+          _.merge(authOpts(), { signal, credentials: 'include' }))
       }
     }
   }
-}
+})
 
 
-export const Dockstore = {
+const Dockstore = signal => ({
   getWdl: async (path, version) => {
-    const res = await fetchDockstore(`${dockstoreMethodPath(path)}/${encodeURIComponent(version)}/WDL/descriptor`)
+    const res = await fetchDockstore(`${dockstoreMethodPath(path)}/${encodeURIComponent(version)}/WDL/descriptor`, { signal })
     return res.json()
   },
 
   getVersions: async path => {
-    const res = await fetchDockstore(dockstoreMethodPath(path))
+    const res = await fetchDockstore(dockstoreMethodPath(path), { signal })
     return res.json()
   }
-
-}
+})
 
 
-export const Martha = {
+const Martha = signal => ({
   call: async uri => {
     return fetchOk(await Config.getMarthaUrlRoot(),
-      _.mergeAll([jsonBody({ uri }), authOpts(), { method: 'POST' }])
+      _.mergeAll([jsonBody({ uri }), authOpts(), { signal, method: 'POST' }])
     ).then(res => res.json())
   }
+})
+
+
+export const Ajax = controller => {
+  const signal = controller && controller.signal
+
+  return {
+    User: User(signal),
+    Groups: Groups(signal),
+    Billing: Billing(signal),
+    Workspaces: Workspaces(signal),
+    Buckets: Buckets(signal),
+    Methods: Methods(signal),
+    Jupyter: Jupyter(signal),
+    Dockstore: Dockstore(signal),
+    Martha: Martha(signal)
+  }
+}
+
+
+export const ajaxCaller = WrappedComponent => {
+  return class AjaxWrapper extends Component {
+    constructor(props) {
+      super(props)
+      this.controller = new window.AbortController()
+      this.ajax = Ajax(this.controller)
+    }
+
+    render() {
+      return h(WrappedComponent, {
+        ref: component => this.child = component,
```

Comment:
Is this being done to support getting refs on the inner component? If so, take a look at `React.forwardRef`